beluga_ros/include/beluga_ros/ndt_ellipsoid.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright 2024 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BELUGA_ROS_NDT_ELLIPSOID_HPP
+#define BELUGA_ROS_NDT_ELLIPSOID_HPP
+
+#include <beluga_ros/messages.hpp>
+
+
+/**
+ * \file
+ * \brief Utilities for NDT I/O over ROS interfaces.
+ */
+
+namespace beluga_ros {
+
+/// 
+template <typename MapType, int NDim>
+//beluga_ros::msg::MarkerArray& assign_obstacle_map(beluga::SparseValueGrid<MapType, NDim> grid){
+beluga_ros::msg::MarkerArray assign_obstacle_map(const beluga::SparseValueGrid<MapType, NDim>& grid){
+
+  // Get data from the grid
+  auto& map = grid.data();
+
+  beluga_ros::msg::MarkerArray message{};
```

Comment:
to remove the previous map you can use:

```suggestion
  beluga_ros::msg::MarkerArray message{};
  {
    beluga_ros::msg::Marker marker;
    marker.ns = "obstacles";
    marker.action = beluga_ros::msg::Marker::REMOVE_ALL;
  }
```